src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -648,6 +648,33 @@ internal static string ReadString(ref Utf8JsonReader reader, string propertyName
             return reader.GetString();
         }
 
+        /// <summary>
+        /// This method allows a JsonTokenType to be string or number but, it will always return it as a string.
```

Comment:
Do we want to add that this is to maintain compatibility with 6x?

Reply:
Yes, also we used to allow 'sub' to be an array and that was failing an older test but, that is out of scope for this one.